astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -85,24 +95,24 @@ def gcrs_precession_mat(equinox):
     return erfa.fw2m(gamb, phib, psib, epsa)
 
 
-def get_location_gcrs(reference, gcrs_to_reference_matrix):
-    """Create a GCRS frame at the location and obstime of a given reference.
+OMEGA_EARTH = u.Quantity(7.292115855306589e-5, 1./u.s)
+"""
+Rotational velocity of Earth. In UT1 seconds, this would be 2 pi / (24 * 3600),
+but we need the value in SI seconds.
+See Explanatory Supplement to the Astronomical Almanac, ed. P. Kenneth Seidelmann (1992),
+University Science Books.
+"""
 
-    Helper function that avoids location.get_gcrs (which would trigger
-    infinite recursion in some cases), and uses an already calculated
-    GCRS to Reference matrix to calculate obsgeoloc and obsgeovel
-    required for GCRS.
-    """
-    # TODO: ideally, GCRS and other frames would share how they describe
-    # the location.  See gh-10996.
+
+def get_location_gcrs(reference, gcrs_to_itrs_matrix):
     obstime = reference.obstime
-    gcrs_cart = (reference.location.get_itrs(obstime, include_velocity=True)
-                 .transform_to(reference.__class__(obstime=obstime))
-                 .cartesian
-                 .transform(matrix_transpose(gcrs_to_reference_matrix)))
-    return GCRS(obstime=obstime,
-                obsgeoloc=gcrs_cart.without_differentials(),
-                obsgeovel=gcrs_cart.differentials['s'].to_cartesian())
+    itrs_cart = reference.location.get_itrs(obstime).cartesian
+    obsgeoloc = itrs_cart.transform(matrix_transpose(gcrs_to_itrs_matrix))
+    vel_x = -OMEGA_EARTH * obsgeoloc.y
```

Comment:
@mkbrewer - is there an ERFA routine that does this (maybe even in a better way)?

Reply:
Hmm, looking through why we changed it in `earth.py`, I came back to #6699, which planned to do:
```
        # CIRS velocity
        rotation_vector = CartesianRepresentation(0/u.s, 0/u.s, OMEGA_EARTH)
        cirs_cart = itrs.transform_to(cirs_frame).cartesian
        cirs_vel = rotation_vector.cross(cirs_cart)
        # Transform to GCRS (rotation)
        obsgeovel = cirs_vel.transform(matrix_transpose(gcrs_to_cirs_mat(obstime)))
        return obsgeoloc, obsgeovel
```
But then in #6861 it was changed to its present state. @StuartLittlefair - both were your PRs, so you probably remember better what is right/wrong with it.

Reply:
This was removed when velocities were added to coordinates purely because it was simpler.